File: app/src/main/java/com/mintthursday/database/RecipeDao.java
package com.mintthursday.database;

import static androidx.room.OnConflictStrategy.*;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.mintthursday.models.Recipe;

import java.util.List;

@Dao
public interface RecipeDao {
    @Query("SELECT * FROM recipe")
    List<Recipe> getAll();

    @Query("SELECT * FROM recipe WHERE id = :id")
    Recipe getById(long id);

    @Insert(onConflict = REPLACE)
    void insert(Recipe recipe);

    @Delete
    void delete(Recipe recipe);
}

File: app/src/main/java/com/mintthursday/database/RecipeDao.kt
package com.mintthursday.database

import androidx.room.*
import com.mintthursday.models.Recipe

@Dao
interface RecipeDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertRecipe(vararg recipe: Recipe)

    @Delete
    fun deleteRecipe(recipe: Recipe?)

    @Query("SELECT * FROM recipe")
    fun getAllRecipes(): List<Recipe>

    @Query("SELECT * FROM recipe WHERE id = :id")
    fun getRecipeById(id: Long): Recipe?

    @Query("SELECT name FROM recipe WHERE id = :id")
    fun getRecipeNameById(id: Long): String
}